lib/src/components/NxToggle/__tests__/NxToggleRTL.test.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import '@testing-library/jest-dom';
+
+import { render, screen, fireEvent } from '@testing-library/react';
+
+import NxToggle, { Props } from '../NxToggle';
+
+describe('NxToggle', function() {
+  const simpleProps: Props = {
+    inputId: 'toggle-id',
+    isChecked: false,
+    onChange: () => {},
+    disabled: undefined,
+    children: 'Enables whales'
+  };
+
+  it('renders a <label> containing a toggle <input> and toggle control', function() {
```

Comment:
This doesn't check for the `<label>` and doesn't check that the `switch` is an `<input>`

Reply:
> This doesn't check for the `<label>` and doesn't check that the `switch` is an `<input>`

Ops,
I added a separate test to check for that below.
Not entirely sure if it's the recommended way in RTL.
I read somewhere that RTL philosophy is to avoid testing implementation details and about testing how the user would interact with the computer. Not entirely sure if this counts the former 🤔
3f77912c8f3e4b30b756cc5e5cb2537c483c65ed

Reply:
> I read somewhere that RTL philosophy is to avoid testing implementation details and about testing how the user would interact with the computer. Not entirely sure if this counts the former thinking

Yeah I'm not sure that philosophy is as valid when testing a component library.  We need to make (and test) guarantees about what HTML elements we're using so that calling code knows what they're constructing and can make determinations about whether it's resulting in valid, semantically-reasonable DOM.

Reply:
> > Yeah I'm not sure that philosophy is as valid when testing a component library. We need to make (and test) guarantees about what HTML elements we're using so that calling code knows what they're constructing and can make determinations about whether it's resulting in valid, semantically-reasonable DOM.
> 
> That said I do get the whole idea of querying by role. What if we refactor that test so that it queries the `switch` role, asserts that the resulting element has a `tagName` of `INPUT`, and then uses the `.labels` property to get the label element and asserts its text content? Asserting the accessible name during the role query would be a good idea too - e.g. `getByRole('switch', { name: 'Enables whales' })`

Thanks that's a good idea. I've updated it here: 2c0f887648cfad445b9a03fb8c3ecc5c2705b501